packages/typescript-estree/tests/lib/convert.test.ts

```diff
@@ -238,6 +238,15 @@ describe('convert', () => {
         type: AST_NODE_TYPES.TSAbstractKeyword,
       });
     });
+
+    it('should throw an error if optional chaining in TaggedTemplateExpression', () => {
+      const ast = convertCode('a?.b``');
+      const instance = new Converter(ast);
+
+      expect(() => instance.convertProgram()).toThrow(
+        'Cannot have an optional chain in TaggedTemplateExpression.',
+      );
+    });
```

Comment:
We don't use this file for AST tests.
Tests in this file are basic validation test to ensure the parser is working as intended.

Instead tests for the AST are hosted within `ast-spec`.

1) open `packages/ast-spec/src/expression/TaggedTemplateExpression`
2) create a new folder called `fixtures/_error_`
3) add `n` folders within `_error_` -- one for each intentionally failing test case
4) in each folder create a `spec.ts` with the failing code
5) run the tests in `ast-spec` to generate the snapshots

Cases we should have which should now be banned:
- `` a?.b`...` ``
- `` a?.b.c`...` ``
- `` a?.b()`...` ``

Worth adding a few just to be sure we have regression coverage

Reply:
Removed it from here and added tests for it in ast-spec.